hana/lib/HANAService.js

```diff
@@ -181,6 +182,20 @@ class HANAService extends SQLService {
     }
   }
 
+  async onNOTFOUND(req, next) {
+    try {
+      return await next()
+    } catch (err) {
+      // TODO: check that the query target is defined in the model
+      // Ensure that the known entity still exists
+      if (/*this.context.tenant && */ err.code === 259) {
```

Comment:
Eventually this ends in a "hick-up" and the current request is terminated with 500. That's the best we can do, isn't it?

Reply:
At this point we have to assume that the database is toast. So it will be required to fail and enable the possibility to recover at a later point in time.